frontend/src/app/hierarchy-viewer/hierarchy-viewer.component.ts

```diff
@@ -19,32 +23,194 @@ export class HierarchyViewerComponent implements OnDestroy {
     subscriptions: Subscription[] = [];
     hierarchies$?: Observable<HierarchyListItem[]>;
 
-    constructor(private store: Store<HierarchyState>) {
+    constructor(public dialog: MatDialog, private store: Store<HierarchyState>, private renderer: Renderer2) {
         const sub = this.store.select(getSelectedHierarchy)
             .subscribe(hierarchy => {
                 if (hierarchy) {
-                    this.selectedHierarchy = hierarchy;              
+                    this.selectedHierarchy = hierarchy;
                 }
             });
         this.subscriptions.push(sub);
-    
+
         this.hierarchies$ = this.store.select(getHierarchies);
```

Comment:
Here we can add the logic to select a hierarchy after deletion. We'll subscribe to the deleteHierarchySuccess action and dispatch a setSelectedHierarchy action. I can do this bit if you'd like.

Reply:
I'll let you do that